Bug 1879663 - Add RFC proposing updates to the RFC process (#5681)
mozilla-mobile/firefox-android: docs/rfcs/0000-template.md

Reviewer: What is the difference between this template and [the previous one](https://mozac.org/rfc/0001-rfc-process)? The prior one worked as a way to introduce the process and be used as a template.

Can we just amend that one with additional information that might be missing?

Author: The value add here would just be around clarity: the new template has detail about what each section is for, the file is clearly named, and has additional sections that have been pulled from other examples. How would we amend the original with additional templating information without diluting the proposal within it?

I also think there is value in treating an accepted RFC as finalized and not updating it in place after the fact, as I've come to learn 😅

Author: > Since it is meant as a template, like our previous case, we can still amend it with annotations.

Is there an example of somewhere we've done that already? My hope is that the template and README become living, update-able documents that are clear entry points to the process. I think if we start amending RFCs, then it becomes less clear which are the most up-to-date, and begs the question of whether they serve as living documentation for the systems and changes they had proposed.

---
layout: page
title: your title
permalink: /docs/rfcs/file-name
---

* RFC PR: [PR #](https://github.com/mozilla-mobile/android-components/pull/#)
* Start date: YYYY-MM-DD (Day of proposal posting.)
* End date: YYYY-MM-DD (Last day for general feedback. However, the proposal can be merged immediately after all stakeholders approve.)
* Stakeholders: github-username, github-username

## Summary

This section should include a brief description of the proposal.

## Motivation

This section should include reasoning about why the proposal is useful. Examples, specific scenarios, 
open bugs, records of performance metrics, and other empirical data are beneficial to include here if available.

## Guide-level explanation

This section should include a high-level walkthrough of the steps required to implement the proposal.

## Reference-level explanation (optional)

This section should include a detailed walkthrough of technical steps or code changes that 
will be required to implement the proposal. Code samples and prototypes are beneficial to include here.

## Drawbacks (optional)

This section should include any drawbacks to the proposal.

> ## Rationale and alternatives (optional)

Reviewer: ```suggestion
## Rationale and alternatives
```

Sorry, I missed this in my previous pass. For the same reason as the comment above on 'Unresolved questions', let's make it required that the developer explicitly says there are no alternatives to their design.

This section should include any alternative proposals considered, as well as the rationale for why
they were not selected for the proposal.

## Resources and Docs (optional)

- Any (internal or external) similar proposals  or other documentation that shares concepts with the proposal.
- Links to artifacts generated as part of the proposal, such as additional documentation or follow-up bugs.

> ## Unresolved questions (optional)

Reviewer: ```suggestion
## Unresolved questions
```

Let's not make this optional. If there are no unresolved questions, then this section would require the developer to be explicit that they considered it and couldn't find any. Otherwise, all optional sections will most likely always be ignored.

---
layout: page

> title: <your title>

Reviewer 2: I think generally we can remove the angled brackets in the file when used to reference 'your thing' as they're fairly self-explanatory? imo it just adds visual noise without much, if any benefit?

permalink: /docs/rfcs/<file-name>
---

> * RFC PR: [<PR #>](https://github.com/mozilla-mobile/android-components/pull/<#>)

Reviewer 2: will this be updated to include a PR number?

* Start date: YYYY-MM-DD (Day of proposal posting.)

> * End date: YYYY-MM-DD (Initial deadline for feedback, subject to blocking changes required. The proposal can be merged immediately after stakeholder approval.)

Reviewer 2: ```suggestion
* End date: YYYY-MM-DD (Last day for general feedback, subject to blocking changes required. The proposal can be merged immediately after stakeholder approval.)
```

> * End date: YYYY-MM-DD (Initial deadline for feedback, subject to blocking changes required. The proposal can be merged immediately after stakeholder approval.)

Reviewer 2: maybe we just call `End Date` `Deadline`?

> * Stakeholders: <github-username>, <github-username>

Reviewer 2: <github-username> doesn't show up in the fie view, is that expected?
<img width="416" alt="Screenshot 2024-02-28 at 10 21 48" src="https://github.com/mozilla-mobile/firefox-android/assets/1833156/a68e0af9-be7e-4a4d-922e-de969d6f9ecb">

---
layout: page
title: <your title>
permalink: /docs/rfcs/<file-name>
---

* RFC PR: [<PR #>](https://github.com/mozilla-mobile/android-components/pull/<#>)
* Start date: YYYY-MM-DD (day of posting)

> * End date: YYYY-MM-DD (initial deadline for feedback, subject to blocking changes required)

Reviewer 3: Do we keep the PRs open until the end date even when approved for others to have the ability to comment?

Author: Updated in a couple places to make clear that proposals can be merged early once they've been approved

* Stakeholders: <github-username>, <github-username>

## Summary

This section should include a brief description of the proposal.

## Motivation

This section should include reasoning about why the proposal is useful. Examples, specific scenarios, 
open bugs, records of performance metrics, and other empirical data are beneficial to include here if available.

## Guide-level explanation

This section should include a high-level walkthrough of the steps required to implement the proposal.

## Reference-level explanation

This optional section should include a detailed walkthrough of technical steps or code changes that 
will be required to implement the proposal. Code samples and prototypes are beneficial to include here.

## Drawbacks

This section should include any drawbacks to the proposal. This section can be considered optional, but is highly encouraged in order to facilitate healthy debate.

## Rationale and alternatives

This section should include any alternative proposals considered, as well as the rationale for why
they were not selected for the proposal. This section can be considered optional, but is highly encouraged in order to facilitate healthy debate.

> ## Prior art

Reviewer 4: I think this and Artifacts can be combined in one section called `Resources and Docs`, wdyt?

Author: I think it seems useful to have separate sections for external resources and artifacts that are being added to the tree as part of the RFC , but I'm open to changing the name. "Prior art" is just borrowed from earlier RFC examples
